File: destack-py/destack/language/deployment/log.py
from typing import TYPE_CHECKING

from destack.language.core import (
    Enum,
    EnumType,
    Event,
    Json,
    NodeType,
    builtin_enum,
    builtin_node,
    builtin_property,
)

if TYPE_CHECKING:
    pass

# pyright: reportIncompatibleVariableOverride=false


@builtin_enum(EnumType.LOG_LEVEL)
class LogLevel(Enum):
    TRACE = 1
    DEBUG = 2
    INFO = 3
    WARNING = 4
    ERROR = 5
    PANIC = 6


@builtin_node(NodeType.LOG_EVENT, frozen=True)
class LogEvent(Event):
    """A Log message."""

    content: str = builtin_property(110)
    attributes: dict[str, Json] = builtin_property(111)
    level: LogLevel = builtin_property(112)

File: destack-rs/destack/src/basics/script/log.rs
//! destack.basics.script.log@2025.08.15.1

#![destack::partial(destack.basics.script.log, file)]

#[destack::generated(LogLevel, enum, block)]
/// LogLevel
pub enum LogLevel {
    /// A Trace
    TRACE = 1,
    /// A Debug
    DEBUG = 2,
    /// An Info
    INFO = 3,
    /// A Warning
    WARNING = 4,
    /// An Error
    ERROR = 5,
    /// A Panic
    PANIC = 6
}